### PR TITLE
[#1166] Add pull request template

### LINKED ISSUE #1166: Setup Github Actions for first time contributors
First-time contributors have a difficult time matching RepoSense's contribution guidelines and it becomes the onus of the senior developers to manually check and guide them. Though this is a good practice, I think we should automate the first level of interaction. 

Suggestion: Setup Github Actions to reply to first-time developers whenever they open a new PR, to guide them related to the checks and commit message. Then, the senior developers can come in and guide. 

Benefits: The senior developers can focus more on guiding the contributor on their code rather than project management.

### PR DESCRIPTION
**Pull request by dcshzj**
Fixes #1166.

## Proposed commit message

```
We follow a certain format for the pull request description, and new
contributors may not necessarily be aware of it until a maintainer
informs them in a comment.

Let's introduce a pull request template that automatically populates the
description box when creating a new pull request, so that new
contributors are aware of the conventions that we follow.
```

## Other information

* Pull request template was based on [the one for MarkBind](https://github.com/MarkBind/markbind/blob/master/.github/PULL_REQUEST_TEMPLATE).
* GitHub has more [documentation on using pull request templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository).
* I added this section to demonstrate how the pull request template would look like, an actual PR only needs this section if the information is not already available in the linked issues.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+-->
+```
+Brief description of current behaviour and reason(s) for the change.
+
+Call-to-action statement describing what you are doing in this PR.
+```
+
+## Other information
+<!--
+    Include any other relevant information, such as special testing
```
Comment (gerhean) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: Feels a little blunt. Referencing the `feature-request.md`, maybe phrasing as question would be nicer:

```
Are there other relevant information, such as special testing instructions, 
which will help the reviewer better understand the code?

You may also include a brief description of why the problem occurred.
```

**Comment by gerhean** [comment]
Furthermore, would it also be a good opportunity to introduce the checkbox like in Markbind?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    If this pull request is part of an issue, but does not fully fix
+    it, use: "Part of #xxxx".
+-->
+Fixes #xxxx
+
+## Proposed commit message
+<!--
+    Propose a detailed commit message for this pull request using the
+    following format. Wrap lines at 72 characters.
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply .github/PULL_REQUEST_TEMPLATE.md]: I was referring to the content in the triple backticks. I modified it to make it a little clearer.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    Guide on how to write a good commit message:
```
[the remark this answers is not in the document]
Comment (dcshzj) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: The most relevant one would be how to write the commit message in a particular way, which I reproduced within the triple backticks.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (dcshzj) [reply .github/PULL_REQUEST_TEMPLATE.md]: Agree, thanks!

**Comment by dcshzj** [comment]
> Furthermore, would it also be a good opportunity to introduce the checkbox like in Markbind?

I think checkboxes might be a bit confusing as GitHub mainly signifies checkboxes as tasks to be completed. I didn't really like how MarkBind was using the checkboxes, as it may seem like the pull request still has some more tasks to be completed before it can be merged.

**Review by gerhean — APPROVED** [review]
LGTM

**Comment by dcshzj** [comment]
Will merge this directly, since it concerns only our workflows and can be adjusted in the future if the need arises.

**Pull request merged**